### PR TITLE
improve xds proxy logs

### PR DESCRIPTION
**Pull request by ramaraochavali**
As discovered https://github.com/istio/istio/issues/35263

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pkg/istio-agent/xds_proxy.go [hunk 1]
```diff
@@ -495,7 +495,7 @@ func (p *XdsProxy) handleUpstreamRequest(con *ProxyConnection) {
 				p.ecdsLastNonce.Store(req.ResponseNonce)
 			}
 			if err := sendUpstream(con.upstream, req); err != nil {
-				proxyLog.Errorf("upstream [%d] send error for type url %s: %v", con.conID, req.TypeUrl, err)
+				proxyLog.Debugf("upstream [%d] send error for type url %s: %v", con.conID, req.TypeUrl, err)
```
Comment (ramaraochavali) [inline pkg/istio-agent/xds_proxy.go:498]: This log is redundant as we already handle upstreamError channel other place

File: pkg/istio-agent/xds_proxy_delta.go [hunk 2]
```diff
@@ -162,7 +162,7 @@ func (p *XdsProxy) handleDeltaUpstream(ctx context.Context, con *ProxyConnection
 		select {
 		case err := <-con.upstreamError:
 			// error from upstream Istiod.
-			if istiogrpc.IsExpectedGRPCError(err) {
+			if istiogrpc.IsExpectedGRPCError(err) || isRateLimited(err) {
```
Comment (ramaraochavali) [inline pkg/istio-agent/xds_proxy_delta.go:165]: considers rate limiting as expected error

**Review by howardjohn — APPROVED** [review]

File: pkg/istio-agent/xds_proxy_delta.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/istio-agent/xds_proxy_delta.go:165]: I am not sure if this is right. In this case, we should let users know what happen, instead of hide it

Comment (ramaraochavali) [reply pkg/istio-agent/xds_proxy_delta.go:165]: But Envoy reconnects like any other gRPC problem like Istiod disconnects every 30 mins. I do not know how is different from that. Istiod disconnects every 30 mins to shed the load.

Comment (hzxuzhonghu) [reply pkg/istio-agent/xds_proxy_delta.go:165]: See https://github.com/istio/istio/pull/40644#issuecomment-1226887934

**Pull request merged**